### ISSUE TITLE
Constants for Double and Float Infinities and NaNs

### ISSUE THREAD

**Issue by praalhans** [Frege/frege#312]
When using the `Double` or `Float` native types, it is not possible (out-of-the-box) to refer to the positive or negative infinity values of either type. Neither is it possible to refer to one particular Not-a-Number (NaN) value.

These constants are defined in Double.POSITIVE_INFINITY and Double.NEGATIVE_INFINITY and Double.NaN. These constants are documented as (in the case for positive infinity in Double):

>  A constant holding the positive infinity of type double. It is equal to the value returned by Double.longBitsToDouble(0x7ff0000000000000L).

However, Frege does not have a definition for this native method longBitsToDouble.

A workaround for this issue is to use the `atod` and `atof` methods defined in `StringJ`. These call the `Double.valueOf` and `Float.valueOf` methods in Java, respectively. Considering the following documentation (in this case double): https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(java.lang.String)

E.g.

```
positiveInfinityD :: Double
positiveInfinityD = String.atod "+Infinity"
negativeInfinityD :: Double
negativeInfinityD = String.atod "-Infinity"
nanD :: Double
nanD = String.atod "NaN"
```

is a valid workaround.

**Comment by praalhans** [comment]
A shorter way to work around this issue is division by zero:

```
posInf :: Double
posInf = 1.0 / 0.0
negInf :: Double
negInf = 1.0 / -0.0
nan :: Double
nan = 0.0 / -0.0
```

**Comment by Ingo60** [comment]
We could also provide

    pure native positiveInfinity  "java.lang.Double.POSITIVE_INFINITY" :: double
    pure native negativeInfinity "java.lang.Double.NEGATIVE_INFINITY" :: double

**Comment by Ingo60** [comment]
We have now

    Double.negativeInfinity, Double.positiveInfinity, Double.nan
    Float.negativeInfinity, Float.positiveInfinity, Float.nan

**Comment by Ingo60** [comment]
fixed with commit     bb7b0df
